### ISSUE TITLE
Using the `phx.gen.live` generator creates failing tests when using the types `:date` and `:time`.

### ISSUE THREAD

**Issue by linusdm** [phoenixframework/phoenix#5077]
### Environment

* Elixir version (elixir -v): `1.14.1-otp-25`
* Phoenix version (mix deps): `master` (1.7)
* Operating system: `MacOS`

### Expected behavior
Passing tests when generating LiveViews and tests when using the `phx.gen.live` generators.

### Actual behavior
Running `mix phx.gen.live Items Item items some_date:date some_time:time` in a new project generates these module attributes in `test/my_app/live/item_live_test.exs`:

```elixir
@create_attrs %{some_date: "2022-11-5", some_time: "14:0"}
@update_attrs %{some_date: "2022-11-6", some_time: "15:1"}
```
Which doesn't pass the tests. Changing these attributes to this, fixes the problems:

```elixir
@create_attrs %{some_date: ~D[2022-11-05], some_time: ~T[14:00:00]}
@update_attrs %{some_date: ~D[2022-11-06], some_time: ~T[15:00:00]}
```
 
I see [some recent changes here](https://github.com/phoenixframework/phoenix/commit/19034199e7a5a90a3373e2bb49f21885e87d0f41#diff-4307581434ae66da4750cde438c2b192c4a11c8b71ba32dbf78aea36daa0ee3e), that might have something to do with this regression.

I guess this might be something to fix in 1.7...

**Comment by Eiji7** [comment]
I can also reproduce this. This should be fixed before `1.7.0` stable release.
ping @chrismccord

**Comment by markmark206** [comment]
@chrismccord / phoenix folks, if this needs a test, and if you are looking for someone to look into that, I'd be happy to investigate.

And in any case, thank you for this awesome framework and ecosystem. ; )

**Comment by josevalim** [comment]
Thanks @Eiji7! Btw, you don't need to ping the maintainers. Maintainers already receive every issue and every comment. We will look at it when we can (unless someone beats us to a PR, which would be great!)

**Comment by paridin** [comment]
Sorry for the dare, I did a PR to solve this issue.

**Comment by jeregrine** [comment]
This was fixed in #5107
